controllers/flinkcluster_updater.go

```diff
@@ -230,15 +231,28 @@ func (updater *ClusterStatusUpdater) deriveClusterStatus(
 			} else {
 				state = v1beta1.ComponentStateNotReady
 			}
+		} else if observedJmService.Spec.Type == corev1.ServiceTypeNodePort {
+			if len(observedJmService.Spec.Ports) > 0 {
+				state = v1beta1.ComponentStateReady
```

Comment:
I think you need to add an `else` statement with `state = v1beta1.ComponentStateNotReady`.